tests/providers/cncf/kubernetes/hooks/test_kubernetes.py

```diff
@@ -37,107 +35,181 @@
-            ('kubernetes_kube_config_empty', {'extra__kubernetes__kube_config': ''}),
-            ('kubernetes_kube_config_path_empty', {'extra__kubernetes__kube_config_path': ''}),
-            ('kubernetes_with_namespace', {'extra__kubernetes__namespace': 'mock_namespace'}),
-            ('kubernetes_default_kube_config', {}),
+            ('in_cluster', {'extra__kubernetes__in_cluster': True}),
+            ('kube_config', {'extra__kubernetes__kube_config': '{"test": "kube"}'}),
+            ('kube_config_path', {'extra__kubernetes__kube_config_path': 'path/to/file'}),
+            ('in_cluster_empty', {'extra__kubernetes__in_cluster': ''}),
+            ('kube_config_empty', {'extra__kubernetes__kube_config': ''}),
+            ('kube_config_path_empty', {'extra__kubernetes__kube_config_path': ''}),
+            ('kube_config_empty', {'extra__kubernetes__kube_config': ''}),
+            ('kube_config_path_empty', {'extra__kubernetes__kube_config_path': ''}),
+            ('context_empty', {'extra__kubernetes__cluster_context': ''}),
+            ('context', {'extra__kubernetes__cluster_context': 'my-context'}),
+            ('with_namespace', {'extra__kubernetes__namespace': 'mock_namespace'}),
+            ('default_kube_config', {}),
         ]:
             db.merge_conn(Connection(conn_type='kubernetes', conn_id=conn_id, extra=json.dumps(extra)))
 
     @classmethod
-    def tearDownClass(cls) -> None:
+    def teardown_class(cls) -> None:
         clear_db_connections()
 
-    @patch("kubernetes.config.incluster_config.InClusterConfigLoader")
-    def test_in_cluster_connection(self, mock_kube_config_loader):
-        kubernetes_hook = KubernetesHook(conn_id='kubernetes_in_cluster')
-        api_conn = kubernetes_hook.get_conn()
-        mock_kube_config_loader.assert_called_once()
-        assert isinstance(api_conn, kubernetes.client.api_client.ApiClient)
-
-    @patch("kubernetes.config.kube_config.KubeConfigMerger")
-    @patch("kubernetes.config.kube_config.KubeConfigLoader")
-    def test_in_cluster_connection_empty(self, mock_kube_config_merger, mock_kube_config_loader):
-        kubernetes_hook = KubernetesHook(conn_id='kubernetes_in_cluster_empty')
-        api_conn = kubernetes_hook.get_conn()
-        mock_kube_config_loader.assert_called_once_with(KUBE_CONFIG_PATH)
-        mock_kube_config_merger.assert_called_once()
-        assert isinstance(api_conn, kubernetes.client.api_client.ApiClient)
-
+    @pytest.mark.parametrize(
+        'in_cluster_param, conn_id, in_cluster_called',
+        (
+            pytest.param(True, None, True),
+            pytest.param(None, None, False),
+            pytest.param(None, 'in_cluster', True),
+            pytest.param(True, 'in_cluster', True),
+            pytest.param(False, 'in_cluster', False),
+            pytest.param(None, 'in_cluster_empty', False),
+            pytest.param(True, 'in_cluster_empty', True),
+            pytest.param(False, 'in_cluster_empty', False),
```

Comment:
```suggestion
            (True, None, True),
            (None, None, False),
            (None, 'in_cluster', True),
            (True, 'in_cluster', True),
            (False, 'in_cluster', False),
            (None, 'in_cluster_empty', False),
            (True, 'in_cluster_empty', True),
            (False, 'in_cluster_empty', False),
```

Nit: these can just be normal tuples.